middle_end/flambda2/terms/apply_expr.ml

```diff
@@ -101,43 +99,22 @@ let [@ocamlformat "disable"] print ppf { callee; continuation; exn_continuation;
 
 let [@ocamlformat "disable"] print_with_cache ~cache:_ ppf t = print ppf t
 
-let invariant env
+let invariant
     ({ callee;
        continuation;
-       exn_continuation;
-       args;
+       exn_continuation = _;
+       args = _;
        call_kind;
-       dbg;
-       inline;
-       inlining_state;
+       dbg = _;
+       inline = _;
+       inlining_state = _;
        probe_name = _
      } as t) =
-  let unbound_continuation cont reason =
-    Misc.fatal_errorf "Unbound continuation %a in %s: %a" Continuation.print
-      cont reason print t
-  in
-  let module E = Invariant_env in
-  Call_kind.invariant env call_kind;
-  (* let stack = E.current_continuation_stack env in *)
-  E.check_simple_is_bound_and_of_kind env callee K.value;
   begin
     match call_kind with
-    | Function (Direct { code_id = _; closure_id = _; return_arity = _ }) ->
-      (* Note that [return_arity] is checked for all the cases below. *)
-      E.check_simples_are_bound env args
-    | Function Indirect_unknown_arity ->
-      E.check_simples_are_bound_and_of_kind env args K.value
-    | Function (Indirect_known_arity { param_arity; return_arity = _ }) ->
-      ignore (param_arity : Flambda_arity.With_subkinds.t);
-      E.check_simples_are_bound env args
-    | Method { kind; obj } ->
-      ignore (kind : Call_kind.method_kind);
-      E.check_simple_is_bound_and_of_kind env obj K.value;
-      E.check_simples_are_bound_and_of_kind env args K.value
+    | Function _ | Method _
```

Comment:
Why is this case shared with the `C_call` actually ? It seems to be the cause of the CI failure

Reply:
Fixed, thanks